src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -2055,194 +2061,195 @@ - (void)_setupSubscriptionWithReason:(NSString *)reason
         });
     }
 
+    // Call directlyGetSessionForNode because the subscription setup already goes through the subscription pool queue
```

Comment:
So I was figuring we would stop putting the subscription setup in there and just call getSessionForNode and have _that_ get queued.... why are we not doing that?

Reply:
Subscriptions on controller startup can cause a thundering herd issue with many Thread devices. If we only use the N-pool for getSessionForNode, then as soon as session is established, another device subscription can move forward, and we can get into a situation where many subscriptions and their priming reports are in the air at the same time.